src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.DocumentActiveContextChangedEventSource.cs

```diff
@@ -11,20 +11,20 @@ internal partial class TaggerEventSources
 {
     private sealed class DocumentActiveContextChangedEventSource(ITextBuffer subjectBuffer) : AbstractWorkspaceTrackingTaggerEventSource(subjectBuffer)
     {
+        private WorkspaceEventRegistration? _documentActiveContextChangedDisposer;
+
         protected override void ConnectToWorkspace(Workspace workspace)
-            => workspace.DocumentActiveContextChanged += OnDocumentActiveContextChanged;
+            => _documentActiveContextChangedDisposer = workspace.RegisterDocumentActiveContextChangedHandler(OnDocumentActiveContextChanged, WorkspaceEventOptions.MainThreadDependent);
```

Comment:
nit: i wouldn't mind docs as to why for anything saying there is a main thread dep.

Reply:
done in all places that currently use the main thread dependency options